fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -696,6 +696,23 @@ public CommandProcessingResult updateClient(final Long clientId, final JsonComma
             }
 
             updateClientAdditionalInfo(clientForUpdate, command);
+            boolean isOdooEnabled = this.configurationDomainService.isOdooIntegrationEnabled();
+            if(isOdooEnabled) {
+                if (changes.containsKey(ClientApiConstants.firstnameParamName) || changes.containsKey(ClientApiConstants.fullnameParamName)
+                        || changes.containsKey(ClientApiConstants.lastnameParamName)
+                        || changes.containsKey(ClientApiConstants.middlenameParamName)
+                        || changes.containsKey(ClientApiConstants.mobileNoParamName)) {
+
+                    boolean status = this.odooService.updateCustomerToOddo(clientForUpdate);
```

Comment:
@Deepika1095  I don't think you need these flags. A client record can be updated more than once. So by applying a flag it means you're allowing it a chance to be updated once .
Please check it out.

Reply:
@hero78 I am maintaining that because when cron job runs it will check which one already has been updated in odoo so don't fetch from database.